pkg/rpc/server/server.go

```diff
@@ -355,20 +370,31 @@ drainloop:
 	}
 }
 
-func (s *Server) handleWsReads(ws *websocket.Conn, resChan chan<- response.Abstract, subscr *subscriber) {
+func (s *Server) handleWsReads(ws *websocket.Conn, resChan chan<- interface{}, subscr *subscriber) {
 	ws.SetReadLimit(wsReadLimit)
 	ws.SetReadDeadline(time.Now().Add(wsPongLimit))
 	ws.SetPongHandler(func(string) error { ws.SetReadDeadline(time.Now().Add(wsPongLimit)); return nil })
 requestloop:
 	for {
-		req := new(request.In)
+		req := request.NewRequest()
 		err := ws.ReadJSON(req)
 		if err != nil {
 			break
 		}
 		res := s.handleRequest(req, subscr)
-		if res.Error != nil {
-			s.logRequestError(req, res.Error)
+		switch {
+		case req.In != nil:
+			r := res.(response.Abstract)
+			if r.Error != nil {
+				s.logRequestError(req.In, r.Error)
+			}
+		case req.Batch != nil:
+			responses := res.([]response.Abstract)
+			for i, r := range responses {
+				if r.Error != nil {
```

Comment:
Some `func RunForErrors(func (*response.Error))` method for an interface can solve this one (avoiding `interface{}` and type casting)